src/main/scala/zio/sqs/SqsPublisher.scala

```diff
@@ -34,4 +38,89 @@ object SqsPublisher {
       })
       ()
     }
+
+  trait Event {
+    def body: String
+    def attributes: Map[String, MessageAttributeValue]
+    def groupId: Option[String]
+    def deduplicationId: Option[String]
+  }
+
+  final case class StringEvent(body: String) extends Event {
+    override def attributes: Map[String, MessageAttributeValue] = Map.empty[String, MessageAttributeValue]
+    override def groupId: Option[String]                        = None
+    override def deduplicationId: Option[String]                = None
+  }
+
+  type ErrorOrEvent = Either[(BatchResultErrorEntry, Event), Event]
+
+  def stringStream(
+    client: SqsAsyncClient,
+    queueUrl: String,
+    settings: SqsPublisherStreamSettings = SqsPublisherStreamSettings()
+  )(
+    ms: Stream[Throwable, String]
+  ): ZStream[Clock, Throwable, ErrorOrEvent] =
+    eventStream(client, queueUrl, settings)(ms.map(StringEvent))
+
+  def eventStream(
+    client: SqsAsyncClient,
+    queueUrl: String,
+    settings: SqsPublisherStreamSettings = SqsPublisherStreamSettings()
+  )(
+    ms: Stream[Throwable, Event]
+  ): ZStream[Clock, Throwable, ErrorOrEvent] =
```

Comment:
How about returning a ZSink instead of a function from stream to stream? A bit like in zio-kafka https://github.com/zio/zio-kafka/blob/master/src/main/scala/zio/kafka/client/Producer.scala#L154

It would feel more natural to combine with an existing stream, what do you think?

Reply:
i though thought at first as well, but there is a nice operator in the current master of zio: [via](https://github.com/zio/zio/pull/2391).

in this case it should be possible to express the streaming logic like:

```
val sendStream = SqsPublisher.stringStream(...)

Stream
   .iterate(..)
   .take(..)
   .map(..)
  .via(sendStream)
  ...
```

Reply:
That looks good as well. That'd be nice to include such an example in the README.

Reply:
yes, you're right. will do it